src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,35 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+    private static LocalDate workingDay;
+    private static LocalDate parseDateFrom;
+    private static LocalDate parseDateTo;
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        parseDateFrom = LocalDate.parse(dateFrom, FORMATTER);
+        parseDateTo = LocalDate.parse(dateTo, FORMATTER);
+        StringBuilder builder = new StringBuilder("Report for period ");
+        builder.append(dateFrom).append(" - ").append(dateTo);
+        for (String name : names) {
+            builder.append(System.lineSeparator()).append(name).append(" - ");
+            int total = 0;
+            for (String line : data) {
+                String[] lines = line.split(" ");
+                workingDay = LocalDate.parse(lines[0], FORMATTER);
```

Comment:
Any magic numbers should be constants. Lets make numbers 0, 1, 2, 3 as constants with informative names